File: python/src/gestor_aplicacion/personas/GrupoHuespedes.py
from gestor_aplicacion.personas.Huesped import Huesped
from gestor_aplicacion.servicios_extra.Factura import Factura
from gestor_aplicacion.hoteles.Habitacion import Habitacion
#from gestor_aplicacion.transporte.Vehiculo import Vehiculo


class GrupoHuespedes:
    def __init__(self, dias_en_hotel, cabeza, habitacion):
        self.dias_en_hotel = dias_en_hotel
        self.cabeza = cabeza
        self.factura = None
        self.habitacion = habitacion
        self.mesa_reservada = None
        self.vehiculo_reservado = None


    def inicializar_factura(self):
        self.factura = Factura()
        self.factura.factura_hospedaje = self.habitacion.precio_x_dia * self.dias_en_hotel

    def __str__(self):
        return 

    def get_dias_en_hotel(self):
        return self.dias_en_hotel

    def set_dias_en_hotel(self, dias_en_hotel):
        self.dias_en_hotel = dias_en_hotel

    def get_lista_huespedes(self):
        return self.lista_huespedes

    def set_lista_huespedes(self, lista_huespedes):
        self.lista_huespedes = lista_huespedes

    def get_factura(self):
        return self.factura

    def set_factura(self, factura):
        self.factura = factura

    def get_habitacion(self):
        return self.habitacion

    def set_habitacion(self, habitacion):
        self.habitacion = habitacion

    def get_mesa_reservada(self):
        return self.mesa_reservada

    def set_mesa_reservada(self, reserva_restaurante):
        self.mesa_reservada = reserva_restaurante

    def get_vehiculo_reservado(self):
        return self.vehiculo_reservado

    def set_vehiculo_reservado(self, vehiculo_reservado):
        self.vehiculo_reservado = vehiculo_reservado

File: src/gestorAplicacion/Personas/GrupoHuespedes.java
package gestorAplicacion.Personas;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import gestorAplicacion.Habitaciones.Habitacion;
import gestorAplicacion.Restaurantes.Mesa;
import gestorAplicacion.serviciosExtra.Factura;
import gestorAplicacion.transporte.Vehiculo;

public class GrupoHuespedes implements Serializable{
    private int diasEnHotel;
    private List<Huesped> listaHuespedes;
    private Factura factura;
    private Habitacion habitacion;

    /*Los atributos mesa y vehiculo se establecen en null por defecto ya que se empiezan a 
     * modificar tras el uso de funcionalidades una vez el grupo de huespedes ya está creado
     */
    private Mesa mesaReservada = null;
    private Vehiculo vehiculoReservado = null;

    /*Sobrecarga de constructores para instanciar objetos con determinada cantidad de parametros */
	public GrupoHuespedes(int diasEnHotel, List<Huesped> listaHuespedes, Habitacion habitacion) {
        this.diasEnHotel = diasEnHotel;
        this.listaHuespedes = listaHuespedes;
        this.habitacion = habitacion;
    }

    public GrupoHuespedes(int diasEnHotel, List<Huesped> listaHuespedes) {
        this.diasEnHotel = diasEnHotel;
        this.listaHuespedes = listaHuespedes;
    }

    public GrupoHuespedes(List<Huesped> listaHuespedes) {
        this.listaHuespedes = listaHuespedes;
    }

    public GrupoHuespedes(){
        this.listaHuespedes = new ArrayList<Huesped>();
    }

    /*agrega un objeto de tipo huesped a la lista de huespedes en un determinado grupo */
    public void agregarHuesped(Huesped huesped){
        listaHuespedes.add(huesped);
    }

    /* inicializa la factura del grupo una vez se crea, solo tiene en cuenta inicialmente el 
     * precio de la habitación y la cantidad de días que se hospeda el grupo
     */
    public void inicializarFactura(){
        factura = new Factura();
        factura.setFacturaHospedaje(habitacion.getPrecioXdia() * diasEnHotel);
    }

    @Override
    public String toString() {
        String lista = "";
        if (listaHuespedes != null){
            for (Huesped huesped : listaHuespedes) {
                if (huesped != null){
                    lista += huesped.toString() + ",";
                }
            }
        }
        //quita coma (,) al final
        if (lista != null && lista.length() > 0 && lista.charAt(lista.length() - 1) == ',') {
            lista = lista.substring(0, lista.length() - 1);
        }
        return lista;
    }

    //* Getters y setters */
    public int getDiasEnHotel() {
        return diasEnHotel;
    }

    public void setDiasEnHotel(int diasEnHotel) {
        this.diasEnHotel = diasEnHotel;
    }

    public List<Huesped> getListaHuespedes() {
        return listaHuespedes;
    }

    public void setListaHuespedes(List<Huesped> listaHuespedes) {
        this.listaHuespedes = listaHuespedes;
    }

    public Factura getFactura() {
        return factura;
    }

    public void setFactura(Factura factura) {
        this.factura = factura;
    }

    public Habitacion getHabitacion() {
        return habitacion;
    }

    public void setHabitacion(Habitacion habitacion) {
        this.habitacion = habitacion;
    }

    public Mesa getMesaReservada() {
        return mesaReservada;
    }

    public void setMesaReservada(Mesa reservaRestaurante) {
        this.mesaReservada = reservaRestaurante;
    }
    
    public Vehiculo getVehiculoReservado() {
 		return vehiculoReservado;
 	}

 	public void setVehiculoReservado(Vehiculo vehiculoReservado) {
 		this.vehiculoReservado = vehiculoReservado;
 	}
    
}
